### ISSUE TITLE
Anchor blcok search and replace does not work with Screen reader

### ISSUE THREAD

**Issue by krperry** [microsoft/vscode#214520]
Type: <b>Bug</b>

With a file like:

hello

hello 
Ken
hello

hello

If you use anchor points like the following  using ctrl+k ctrl+b to drop anchor and ctrl+k ctrl+k to drop end anchor

hello
#drop anchor
hello
ken
hello
#drop end anchor
hello

Then if you do ctrl+h to search and replace. put hel as the text to replace and jel to replace . Then check the replac ein selection. 
finally do replace all.

What you end up with is:
 hello

hello
ken
hello
jello

This happens with all windows screen readers. Jaws, NVDA, and Narrirator.  It seems the selction goes away as soon as you try to search and replace.

VS Code version: Code 1.90.0 (89de5a8d4d6205e5b11647eb6a74844ca23d2573, 2024-06-04T19:33:54.889Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2904)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.88GB (23.64GB free)|
|Process Argv|stm.py --crash-reporter-id dc52a40a-b23f-401d-8926-5a03aba1d0ec|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (16)</summary>

Extension|Author (truncated)|Version
---|---|---
copilot|Git|1.200.0
copilot-chat|Git|0.16.0
elixir-ls|Jak|0.21.3
black-formatter|ms-|2024.2.0
debugpy|ms-|2024.6.0
python|ms-|2024.8.0
vscode-pylance|ms-|2024.6.1
remote-wsl|ms-|0.88.2
cmake-tools|ms-|1.18.41
cpptools|ms-|1.20.5
cpptools-extension-pack|ms-|1.3.0
powershell|ms-|2024.2.2
vsliveshare|ms-|1.0.5918
AudioQ|Seh|0.0.2
lua|sum|3.9.2
cmake|twx|0.0.17

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
pythoncet0:30885854
2e7ec940:31000449
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelc:31048052
dsvsc021:30996838
0ee40948:31013168
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupyter:31046869
26j00206:31048877

```

</details>

**Comment by meganrogge** [comment]
I'm finding this issue hard to follow. Could you pls create a simple reproducible case and describe the steps to take in sequence and what you expect to happen vs what actually happens?
